lib/spack/env/cc

```diff
@@ -297,17 +396,21 @@ input_command="$*"
 # The libs variable is initialized here for completeness, and it is also
 # used later to inject flags supplied via `ldlibs` on the command
 # line. These come into the wrappers via SPACK_LDLIBS.
-#
-includes=()
-libdirs=()
-rpaths=()
-system_includes=()
-system_libdirs=()
-system_rpaths=()
-libs=()
-other_args=()
-isystem_system_includes=()
-isystem_includes=()
+
+# these are colon-separated arrays of directories
+include_dirs_list=""
+lib_dirs_list=""
+rpath_dirs_list=""
+system_include_dirs_list=""
+system_lib_dirs_list=""
+system_rpath_dirs_list=""
+isystem_system_include_dirs_list=""
+isystem_include_dirs_list=""
```

Comment:
Oh right. The comment is old. Yes they are very much intended to be bell-separated, as it’s the most general type of list.

These need to be bell-separated because their contents can come from the command line. The only things that are really guaranteed to be colon-safe are our own env vars, mainly because we made them on the spack side (though we don’t check this). We might consider bellifying our env vars but I think it’s worth making them colon-separated for readability.

Reply:
edited comment